packages/typescript-group-checker/src/compilers/compiler-with-watch.ts

```diff
@@ -0,0 +1,154 @@
+import path from 'path';
+
+import ts, { textChangeRangeIsUnchanged } from 'typescript';
+import { propertyPath, Task } from '@stryker-mutator/util';
+import { StrykerOptions } from '@stryker-mutator/api/core';
+import { tokens, commonTokens, PluginContext, Injector } from '@stryker-mutator/api/plugin';
+import { Logger } from '@stryker-mutator/api/logging';
+
+import { DependencyFile, TypescriptCompiler } from '../compiler';
+
+import { MemoryFileSystem } from '../fs/memory-filesystem';
+import { determineBuildModeEnabled, guardTSVersion, overrideOptions, retrieveReferencedProjects, toPosixFileName } from '../fs/tsconfig-helpers';
+import * as pluginTokens from '../plugin-tokens';
+
+export class CompilerWithWatch implements TypescriptCompiler {
+  public static inject = tokens(commonTokens.logger, commonTokens.options, pluginTokens.mfs);
+
+  private readonly allTSConfigFiles: Set<string>;
+  private readonly tsconfigFile: string;
+  private currentTask = new Task();
+  private currentErrors: ts.Diagnostic[] = [];
+  private readonly sourceFiles: DependencyFile[] = [];
+
+  constructor(private readonly log: Logger, private readonly options: StrykerOptions, private readonly fs: MemoryFileSystem) {
+    this.tsconfigFile = toPosixFileName(this.options.tsconfigFile);
+    this.allTSConfigFiles = new Set([path.resolve(this.tsconfigFile)]);
+  }
+
+  public async init(): Promise<{ dependencyFiles: DependencyFile[]; errors: ts.Diagnostic[] }> {
```

Comment:
Kan deze init methode opgesplitst worden in meerdere kleine (beschrijvende) methodes? Hij is nu iets van 90 regels lang